hyasinsight_connector.py

```diff
@@ -174,12 +174,45 @@ def get_flatten_json_response(self, raw_api_response):
 
         return flatten_json_response
 
+    def _get_error_message_from_exception(self, error):
+        """This function is used to get appropriate error message from the
+        exception.
+        :param e: Exception object
+        :return: error message
+        """
+        try:
+            if error.args:
+                if len(error.args) > 1:
+                    error_code = error.args[0]
+                    error_msg = error.args[1]
+                elif len(error.args) == 1:
+                    error_code = HYAS_ERR_CODE_MSG
+                    error_msg = error.args[0]
+            else:
+                error_code = HYAS_ERR_CODE_MSG
+                error_msg = HYAS_ERR_MSG_UNAVAILABLE
+        except:
+            error_code = HYAS_ERR_CODE_MSG
+            error_msg = HYAS_ERR_MSG_UNAVAILABLE
+
+        try:
+            if error_code in HYAS_ERR_CODE_MSG:
+                error_text = f"Error Message: {error_msg}"
+            else:
+                error_text = f"Error Code: {error_code}. Error Message: " \
+                             f"{error_msg}"
+
+        except:
+            error_text = HYAS_PARSE_ERR_MSG
+
+        return error_text
+
     def _make_rest_call(
             self, endpoint, action_result, data=None, headers=None,
             method="post"
     ):
         try:
-            request_func = getattr(requests, method, timeout=DEFAULT_REQUEST_TIMEOUT)
```

Comment:
@RamboV, As per standard practice we have to put default timeout otherwise the pipeline will fail.
Can you revert this? 
You can increase the timeout if you want.

Reply:
@mishalp-crest I have added the timeout.